source/demo-tasks/simple_sub_pub_demo.c

```diff
@@ -211,6 +211,17 @@ extern MQTTAgentContext_t xGlobalMqttAgentContext;
     static char topicBuf[ 1U ][ mqttexampleSTRING_BUFFER_LENGTH ];
 #endif
 
+/**
+ * @brief Pass and fail counts for each created task.
+ */
+#if democonfigNUM_SIMPLE_SUB_PUB_TASKS_TO_CREATE > 0
+    static volatile uint32_t ulQoS0FailCount[ democonfigNUM_SIMPLE_SUB_PUB_TASKS_TO_CREATE ] = { 0UL }, ulQoS1FailCount[ democonfigNUM_SIMPLE_SUB_PUB_TASKS_TO_CREATE ] = { 0UL };
+    static volatile uint32_t ulQoS0PassCount[ democonfigNUM_SIMPLE_SUB_PUB_TASKS_TO_CREATE ] = { 0UL }, ulQoS1PassCount[ democonfigNUM_SIMPLE_SUB_PUB_TASKS_TO_CREATE ] = { 0UL };
```

Comment:
Since each task uses only either QoS 0 or QoS 1, this is twice what's needed. You could save some space if desired and index using `ulTaskNumber >> 1`:
```suggestion
    static volatile uint32_t ulQoS0FailCount[ ( democonfigNUM_SIMPLE_SUB_PUB_TASKS_TO_CREATE + 1 ) / 2 ] = { 0UL }, ulQoS1FailCount[ ( democonfigNUM_SIMPLE_SUB_PUB_TASKS_TO_CREATE + 1 ) / 2 ] = { 0UL };
    static volatile uint32_t ulQoS0PassCount[ ( democonfigNUM_SIMPLE_SUB_PUB_TASKS_TO_CREATE + 1 ) / 2 ] = { 0UL }, ulQoS1PassCount[ ( democonfigNUM_SIMPLE_SUB_PUB_TASKS_TO_CREATE + 1 ) / 2 ] = { 0UL };
```